apps/prairielearn/src/components/Navbar.html.tsx

```diff
@@ -8,6 +10,15 @@ import { IssueBadge } from './IssueBadge.html.js';
 import type { NavbarType, NavPage, NavSubPage } from './Navbar.types.js';
 import { ContextNavigation } from './NavbarContext.html.js';
 
+export function PreactNavbar(props: {
+  resLocals: Record<string, any>;
+  navPage?: NavPage;
+  navSubPage?: NavSubPage;
+  navbarType?: NavbarType;
+}) {
+  return <div dangerouslySetInnerHTML={{ __html: Navbar(props).toString() }}></div>;
```

Comment:
Here's an example of making a non-JSX component available in JSX. It's not pretty and requires a wrapping element (see https://github.com/preactjs/preact/discussions/4311), so we'd probably want to switch to authoring things in JSX and rendering to an HTML string when needed.